app/src/ui/add-repository/create-repository.tsx

```diff
@@ -149,6 +173,11 @@ export class CreateRepository extends React.Component<
     this.setState({ isRepository, path })
   }
 
+  private async doesReadMeExist(path: string, name: string) {
```

Comment:
Seeing how this method doesn't use any of the local state in the class I think we should either move it outside of the class and turn it into a stateless function or (and I think this would be more in line with what we've done elsewhere) rename it to `updateReadmeExists` and have the method itself call `setState`. Multiple calls to `setState` won't trigger individual updates so there shouldn't be a performance difference.

Reply:
Sounds great, @niik! I will make both changes, and for this specific one, I'll give the `updateReadmeExists` suggestion a try. Thanks for the feedback!